components/tcf/firstLayer/src/index.js

```diff
@@ -1,114 +1,204 @@
-import React, {useState, useEffect} from 'react'
+import React, {useState, useEffect, useRef, useCallback} from 'react'
 import PropTypes from 'prop-types'
 import SuiButton from '@s-ui/react-atom-button'
 import SuiModal from '@s-ui/react-molecule-modal'
-import IconClose from './components/IconClose'
-import InfoCard from './components/InfoCard'
+import SuiNotification from '@s-ui/react-molecule-notification'
+import IconClose from './iconClose'
+import {I18N} from './settings'
+
+const CLASS = 'sui-TcfFirstLayer'
 
 export default function TcfFirstLayer({
-  loadUserConsent,
+  lang = 'es',
+  logo,
+  isMobile,
+  isTablet,
   getVendorList,
   saveUserConsent,
-  openSecondLayer
+  openSecondLayer,
+  openCookiepolicyLayer
 }) {
-  const [saveButtonDisabled, setSaveButtonDisabled] = useState(true)
-  const [state, setState] = useState(null)
-  const [vendorListState, setVendorListState] = useState(null)
-  const [modalOpen, setModalOpen] = useState(true)
+  const [show, setShow] = useState(true)
+  const i18n = I18N[lang]
 
-  useEffect(() => {
-    const loadConsent = async () => {
-      const {purpose} = await loadUserConsent()
-      setState({purposes: purpose})
+  const handleCookiePolicyLayerClick = useCallback(() => {
+    cookiesPolicyLink.current &&
+      cookiesPolicyLink.current.removeEventListener(
+        'click',
+        handleCookiePolicyLayerClick
+      )
+    openCookiepolicyLayer()
+  })
+  const cookiesPolicyLink = useRef()
+  const ref = useRef(null)
+  const textRef = useCallback(
+    node => {
+      if (ref.current) {
+        ref.current = null
+      }
+
+      if (node) {
+        cookiesPolicyLink.current = node.querySelector(
+          `.${CLASS}-body-info-link`
+        )
+        if (cookiesPolicyLink) {
+          cookiesPolicyLink.current.addEventListener(
+            'click',
+            handleCookiePolicyLayerClick
+          )
+        }
+        ref.current = node
+      }
+    },
+    [handleCookiePolicyLayerClick]
+  )
+
+  let initialYOffset
+  let count = 0
+  const checkScroll = () => {
+    count++
+    // use counter to avoid initial window.pageYOffset pointing erroneously to 0
+    if (count === 3) {
+      initialYOffset = window.pageYOffset
     }
-    const loadVendorList = async () => {
-      const {purposes} = await getVendorList()
-      setVendorListState({purposes})
+    if (
+      (count >= 3 && initialYOffset - window.pageYOffset >= 250) ||
+      initialYOffset - window.pageYOffset <= -250
+    ) {
+      handleSaveExitClick()
     }
-    loadConsent()
-    loadVendorList()
-  }, [getVendorList, loadUserConsent])
+  }
+  useEffect(() => {
+    document.addEventListener('scroll', checkScroll)
```

Comment:
maybe using a passive event listener would be better

See https://github.com/WICG/EventListenerOptions/blob/gh-pages/explainer.md
And https://caniuse.com/#feat=passive-event-listener

Reply:
If I've understood well the documentation it seems that this passive option has to be used for events that can delay the scrolling response, not for the scroll event itself. 

`By marking a touch or wheel listener as passive, the developer is promising the handler won't call preventDefault to disable scrolling.`

Don't you understand it this way?

Reply:
Well, all non-passive listeners delay the whatever thing they're listening to, as they cannot know if the listener will cancel the event propagation by calling the `preventDefault`, so they must always wait for listener execution to complete.
With passive event listeners, the "listener code" is ran (in some moment), but without blocking.
This is so small code, so it's OK for me to go as it is 👍  (just seemed nice to me to try this xD)

Reply:
Done! thanks